karl/textindex.py

```diff
@@ -84,3 +94,186 @@ def max_ranked(self):
             max_ranked = int(get_setting(self, 'pgtextindex.max_ranked', 6000))
             self._v_max_ranked = max_ranked
         return max_ranked
+
+    def drop_and_create(self):
+        cm = self.connection_manager
+        conn = cm.connection
+        cursor = cm.cursor
+        try:
+            # Create the table.
+            stmt = """
+            DROP TABLE IF EXISTS %(table)s;
+
+            CREATE TABLE %(table)s (
+                docid INTEGER NOT NULL PRIMARY KEY,
+                community_docid varchar(100),
```

Comment:
Probably should also be ``bigint``, given that this column contains ZODB keys as its values.  However, if it works as is, or unsure whether my assumption is right, this should be postponed until further investigation.